spacy_llm/tasks/ner/task.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Callable, Dict, Iterable, List, Optional, Type
+
+from spacy.language import Language
+from spacy.scorer import get_ner_prf
+from spacy.tokens import Doc, Span
+from spacy.training import Example
+from spacy.util import filter_spans
+
+from ...compat import Literal
+from ...ty import TaskResponseParserProtocol
+from ..span import SpanExample, SpanTask
+from ..templates import read_template
+
+DEFAULT_NER_TEMPLATE_V1 = read_template("ner.v1")
+DEFAULT_NER_TEMPLATE_V2 = read_template("ner.v2")
+
+
+class NERTask(SpanTask):
+    def __init__(
+        self,
+        labels: List[str],
+        template: str,
+        parse_responses: TaskResponseParserProtocol,
+        fewshot_example_type: Type[SpanExample],
```

Comment:
This type can't be `SpanExample` because `SpanExample` assumes the `text`, and `entities` fields. The COT prompting has different types for the keys and wouldn't be compatible.

Reply:
In this case `SpanTask` will have to be modified as well, as it relies on `entities` and `text` being attributes of `SpanExample`. I suggest approaching this in the CoT PR though.